pkg/resources/view_test.go

```diff
@@ -0,0 +1,355 @@
+				},
+			},
+			InputPolicyReferences: []sdk.PolicyReference{},
+			Expected: map[string]any{
+				"column": []map[string]any{
+					{
+						"column_name": "name",
+						"comment":     "comment",
+					},
+					{
+						"column_name": "name2",
+						"comment":     "comment2",
+					},
+				},
+			},
+		},
+		{
+			InputColumns: []sdk.ViewDetails{
+				{
+					Name:    "name",
+					Comment: sdk.String("comment"),
+				},
+				{
+					Name:    "name2",
+					Comment: sdk.String("comment2"),
+				},
+			},
+			InputPolicyReferences: []sdk.PolicyReference{
+				{
+					PolicyDb:      sdk.String("db"),
+					PolicySchema:  sdk.String("sch"),
+					PolicyName:    "policyName",
+					PolicyKind:    sdk.PolicyKindProjectionPolicy,
+					RefColumnName: sdk.String("name"),
+				},
+			},
+			Expected: map[string]any{
+				"column": []map[string]any{
+					{
+						"column_name": "name",
+						"comment":     "comment",
+						"projection_policy": []map[string]any{
+							{
+								"policy_name": sdk.NewSchemaObjectIdentifier("db", "sch", "policyName").FullyQualifiedName(),
+							},
+						},
+					},
+					{
+						"column_name": "name2",
+						"comment":     "comment2",
+					},
+				},
+			},
+		},
+		{
+			InputColumns: []sdk.ViewDetails{
+				{
+					Name:    "name",
+					Comment: sdk.String("comment"),
+				},
+				{
+					Name:    "name2",
+					Comment: sdk.String("comment2"),
+				},
+			},
+			InputPolicyReferences: []sdk.PolicyReference{
+				{
+					PolicyDb:      sdk.String("db"),
+					PolicySchema:  sdk.String("sch"),
+					PolicyName:    "policyName",
+					PolicyKind:    sdk.PolicyKindProjectionPolicy,
+					RefColumnName: sdk.String("name"),
+				},
+				{
+					PolicyDb:          sdk.String("db"),
+					PolicySchema:      sdk.String("sch"),
+					PolicyName:        "policyName2",
+					PolicyKind:        sdk.PolicyKindMaskingPolicy,
+					RefColumnName:     sdk.String("name"),
+					RefArgColumnNames: sdk.String("[one,two]"),
+				},
+			},
+			Expected: map[string]any{
+				"column": []map[string]any{
+					{
+						"column_name": "name",
+						"comment":     "comment",
+						"projection_policy": []map[string]any{
+							{
+								"policy_name": sdk.NewSchemaObjectIdentifier("db", "sch", "policyName").FullyQualifiedName(),
+							},
+						},
+						"masking_policy": []map[string]any{
+							{
+								"policy_name": sdk.NewSchemaObjectIdentifier("db", "sch", "policyName2").FullyQualifiedName(),
+								"using":       []string{"name", "one", "two"},
+							},
+						},
+					},
+					{
+						"column_name": "name2",
+						"comment":     "comment2",
+					},
+				},
+			},
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("handle columns(%d): %v - %v", i, tc.InputColumns, tc.InputPolicyReferences), func(t *testing.T) {
+			valueSetter := newTestResourceValueSetter()
+			err := handleColumns(valueSetter, tc.InputColumns, tc.InputPolicyReferences)
+
+			if tc.Error != "" {
```

Comment:
there is no test case with Error,please remove it (or add err case if needed)

Reply:
Removed